src/logic/calculate.js

```diff
@@ -0,0 +1,56 @@
+import operate from './operate';
+
+const calculate = (calcData, buttonName) => {
+  let { total, next, operation } = calcData;
+  const operators = ['+', '-', '÷', 'X', '%'];
+  if (buttonName === 'AC' || total === '∞') {
+    total = null;
+    next = null;
+    operation = null;
+  } else if (buttonName === '=') {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      next = null;
+      operation = null;
+    }
+  } else if (operators.includes(buttonName)) {
+    if (buttonName === '%') {
+      if (!next) {
+        next = total;
+      }
+      operation = buttonName;
+      total = operate(null, next, operation);
+      next = null;
+      operation = null;
```

Comment:
- [ ] The modifier operate should turn the number into a percentage based number, e.g. `95 % == .95`. We also need to move it to the operate module.
![Screenshot from 2021-03-11 09-22-16](https://user-images.githubusercontent.com/30977845/110786791-53803000-824b-11eb-9242-00e79a6a7304.png)

Reply:
Done